### DISCUSSION TITLE
2022-04-01 Developer Tools Update

### DISCUSSION THREAD

**Discussion by austinbaggio** [near/community#240] [Developer Tools]
The Dev Tools will be combining the effort of both the Core Tools (Rust) & Frontier Tools (JS) teams. Together, we will be focused on NEAR's tools, APIs, & SDKs for both Rust and Javascript. In the near future, we will include the JS updates here. 

## Tools Feedback 
You can reach out to Austin Baggio (Tools Product Manager) to share any feedback about this update or your experience using any of our Dev Tools:  
[30m Calendly Timeslot](https://calendly.com/austinbaggio/30min) | austin.baggio@near.org | [Discord](https://discord.com/channels/490367152054992913/542945453533036544) | [Twitter](https://twitter.com/AustinBaggio)

### nearcore
- @austinabell Finalized nearcore receipt handling refactor https://github.com/near/nearcore/pull/6468
- @miraclx Fixed the nearcore CI pipeline https://github.com/near/nearcore/pull/6512.

### core-contracts
- @austinabell Implemented state replacement contract, optimized state cleanup contract, and e2e prop tested both https://github.com/near/core-contracts/pull/208

### workspaces-rs
- @ChaoticTempest implemented time-travel (fast-forwarding), the last remaining large feature of Workspaces 0.2 https://github.com/near/workspaces-rs/pull/73
- @ChaoticTempest moved credential storage to the user side for workspaces https://github.com/near/workspaces-rs/pull/98 
- @ChaoticTempest fixed some transactions failing from waiting on sandbox to startup https://github.com/near/workspaces-rs/pull/99

### near-sdk-rs
- @austinabell Deprecated and removed old proc macro attributes https://github.com/near/near-sdk-rs/pull/770 & Moved macro generated types into private module https://github.com/near/near-sdk-rs/pull/771
- @austinabell Updated the changelog format https://github.com/near/near-sdk-rs/pull/773
- @itegulov Researched ACI schema formats and code generation

**COMMUNITY FEEDBACK REQUIRED** 
- @austinabell Investigated payable defaults within SDK and inconsistent behavior, considering a proposal for protocol change https://github.com/near/near-sdk-rs/pull/774

### near-jsonrpc-client
- @miraclx Refactoring the gas_price RPC method struct to fix serialization issues without breaking backward compatibility.

**Comment by austinbaggio** [comment]
## near-sdk-js
- @ailisp Added @NearBindgen decorators and code generation of exporting contract method features, automatic serialization/deserialization of contract state, and arguments mechanism. As a result, a high-level JS contract can be written with less code than an equivalent Rust contract.
- @ailisp Added a high-level counter & status-message example https://github.com/near/near-sdk-js/tree/master/examples

## near-workspaces-js
- @ailisp Fixed the near-sdk-js tests with the workspaces-js 1.1 release. Tests can now be run without manual patches
- @volovyk-s & @ailisp Planned scope and delivery plan for the workspaces-js 2.0 release

**Reply by maxzaver** [reply]
s/near-api-js/near-sdk-js

**Comment by maxzaver** [comment]
@AustinBaggio 

> @austinabell Finalized nearcore receipt handling refactor https://github.com/near/nearcore/pull/6468

It seems like PR is still WIP and will be potentially split into multiple PRs.

**Reply by austinabell** [reply]
> @AustinBaggio
> 
> > @austinabell Finalized nearcore receipt handling refactor [near/nearcore#6468](https://github.com/near/nearcore/pull/6468)
> 
> It seems like PR is still WIP and will be potentially split into multiple PRs.

The pattern we've done for community updates is including things when they are done, not when they are pulled in, are you suggesting that we switch to include things that were merged in that week?

The PR is no longer WIP and just needs to resolve reviews. I'll update the description when I get to updating as it mentions that it's WIP
